### ISSUE TITLE
io.js support

### ISSUE THREAD

**Issue by Martin-Pitt** [LinusU/fs-xattr#7]
I am trying to `npm install appdmg`, but this module keeps failing at configuration on my system.
Have got io.js installed and have Mac OSX.

Is io.js not supported?

npm-debug.log
```
956 verbose stack Error: fs-xattr@0.1.5 install: `node-gyp rebuild`
956 verbose stack Exit status 1
956 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:213:16)
956 verbose stack     at emitTwo (events.js:87:13)
956 verbose stack     at EventEmitter.emit (events.js:169:7)
956 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:14:12)
956 verbose stack     at emitTwo (events.js:87:13)
956 verbose stack     at ChildProcess.emit (events.js:169:7)
956 verbose stack     at maybeClose (child_process.js:984:16)
956 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1057:5)
957 verbose pkgid fs-xattr@0.1.5
958 verbose cwd /Users/martin/Projects/Client/Brand/Project
959 error Darwin 14.1.0
960 error argv "node" "/usr/local/bin/npm" "install" "--save-dev" "appdmg"
961 error node v1.2.1-nightly201502201bf91878e7
962 error npm  v2.5.1
963 error code ELIFECYCLE
964 error fs-xattr@0.1.5 install: `node-gyp rebuild`
964 error Exit status 1
965 error Failed at the fs-xattr@0.1.5 install script 'node-gyp rebuild'.
965 error This is most likely a problem with the fs-xattr package,
965 error not with npm itself.
965 error Tell the author that this fails on your system:
965 error     node-gyp rebuild
965 error You can get their info via:
965 error     npm owner ls fs-xattr
965 error There is likely additional logging output above.
```

**Comment by LinusU** [comment]
I haven't tried this on `io.js` but it's to my understanding that it breaks almost all compiled modules. Your log doesn't contain any useful output, could you clone this repo and run `npm install`?

**Comment by stephan-nordnes-eriksen** [comment]
+1 I got the same/similar issue on windows. I cloned the current github version and ran npm install. I got the following:

```
C:\Users\Me\Documents\GitHub\fs-xattr>npm install

> fs-xattr@0.1.5 install C:\Users\Me\Documents\GitHub\fs-xattr
> node-gyp rebuild


C:\Users\Me\Documents\GitHub\fs-xattr>if not defined npm_config_node_gyp
(node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\no
de_modules\node-gyp\bin\node-gyp.js" rebuild )  else (rebuild)
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  xattr.cc
..\src\xattr.cc(7): fatal error C1083: Cannot open include file: 'sys/xattr.h': No such file or directory [C:\Users\S tephan_2\Documents\GitHub\fs-xattr\build\xattr.vcxproj]
gyp ERR! build error
gyp ERR! stack Error: `msbuild` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\node-gyp\lib\build.js:269:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "node" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Me\Documents\GitHub\fs-xattr
gyp ERR! node -v v0.12.2
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok

npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.4
npm ERR! code ELIFECYCLE
npm ERR! fs-xattr@0.1.5 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fs-xattr@0.1.5 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the fs-xattr package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls fs-xattr
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Me\Documents\GitHub\fs-xattr\npm-debug.log
```
npm -v: 2.7.4
node -v: v0.12.2
windows 8.1

**Comment by LinusU** [comment]
@stephan-nordnes-eriksen Ehh, windows dosen't support extended attributes on its filesystems at all so I'm not surprised that that isn't working.

Here is the discussion for [windows support for a similar package for python](https://github.com/xattr/xattr/issues/14). My standpoint is about the same, namely:

> xattr is currently a thin wrapper over a very specific C API. I'm not sure that it would be a good idea to add support for other APIs that are totally different even if they could be used in similar ways.

**Comment by stephan-nordnes-eriksen** [comment]
@LinusU Alright, I didn't know. Sounds good. Probably should find whatever npm module using this and help them get it running on Windows without your library :)

**Comment by LinusU** [comment]
The only package listed on npm with `fs-xattr` as a dependency is `appdmg`, which is why I made this in the first place. It only runs on Mac OS X so for that reason it isn't a problem.

**Comment by LinusU** [comment]
This should be fixed by version `0.1.6`, please try it with that one.
